zomes/chat/src/pagination_helper.rs

```diff
@@ -0,0 +1,106 @@
+//! TODO: Document how to use this crate in general
+//!
+//!
+//!
+use crate::{error::ChatResult, ChatError};
+use chrono::{DateTime, Datelike, NaiveDateTime, Timelike, Utc};
+use hdk::prelude::*;
+use std::cmp;
+
+pub fn get_message_links(
+    channel: Path,
+    earliest_seen: Option<Timestamp>,
+    target_count: usize,
+) -> Result<Vec<Link>, WasmError> {
+    let newest_included_hour = if let Some(earliest_seen) = earliest_seen {
+        if let Ok(hour) = earliest_seen - std::time::Duration::from_secs(60 * 60) {
+            hour
+        } else {
+            return Ok(Vec::new());
+        }
+    } else {
+        sys_time()?
+    };
+
+    let mut links = Vec::new();
+
+    let root_path_length = channel.as_ref().len();
+    let newest_included_hour_path = timestamp_into_path(channel, newest_included_hour)?;
+    if newest_included_hour_path.exists()? {
+        links.append(&mut get_links(newest_included_hour_path.hash()?, None)?.into_inner());
+    }
+
+    let mut earliest_seen_child_path = newest_included_hour_path;
+    let mut current_search_path = earliest_seen_child_path.parent().unwrap();
+    let mut depth = 0;
```

Comment:
The term depth here is confusing because it's actually anti-depth or something like that.